templates/algob-react-boilerplate/.eslintrc.js

```diff
@@ -0,0 +1,51 @@
+module.exports = {
+  env: {
+    browser: true,
+    node: true,
+    es6: true,
+    es2020: true,
+    mocha: true
+  },
+  plugins: ["simple-import-sort", "sonarjs"],
+  extends: [
+    "plugin:sonarjs/recommended",
+    'airbnb',
+    "eslint:recommended", // recommended rules
+  ],
+  parser: "babel-eslint", // for react
+  parserOptions: {
+    ecmaVersion: 2020,
+    // sourceType: "module"
+  },
+  rules:  {
+    "import/no-extraneous-dependencies": 0,
+    "import/prefer-default-export": "off",
+    "max-classes-per-file": 0,
+    "max-len": ["error", { "code": 110, "ignoreTrailingComments": true, "ignoreStrings": true, "ignoreUrls": true, "ignoreTemplateLiterals": true}],
+    "no-underscore-dangle": 0,
+    "simple-import-sort/imports": "warn",
+    "sort-imports": "off",
+    "sonarjs/cognitive-complexity": ["error", 16],
+    "no-unused-vars": ["warn", { "vars": "all", "args": "after-used", "ignoreRestSiblings": false }],
+    "quotes": "off",
+    "strict-boolean-expressions": "off",
+    "react/jsx-filename-extension": "off",
+    "react/react-in-jsx-scope": "off",
```

Comment:
why using this custom rules?

Reply:
Removed `react/jsx-filename-extension` after adding plugin "react/recommended".

For rule `react/react-in-jsx-scope` (line 33), if i don't turn this off, we need to import React in each jsx file (`import React from "react")`. Linter would throw error otherwise.
![image](https://user-images.githubusercontent.com/33264364/110368553-37279c00-806f-11eb-8783-aa08aa64fcfe.png)

Reply:
i think we should import it in jsx file because https://reactjs.org/docs/jsx-in-depth.html
the rule suggests to do so, when we add button etc,, we will need it.

Reply:
Yes, but shouldn't we import this when we need it (like adding a button as you said) ? If i don't turn this off, we'll need to add `import React from "react"` in each `.js` file (even when we aren't using any react component also). I wouldn't want to import this line in files where i don't need this (say if i am using `material-ui` for components, and not react native components). 

If you still think this should be added then i'll do it. Please let me know

Reply:
The documentation says:
> Since JSX compiles into calls to React.createElement, the React library must also always be in scope from your JSX code.

So, let's don't trick the linter, and import it. I think `plugin:react/recommended` should be good to respect.

Reply:
Alright